rllib/core/models/tests/test_mlp_encoders.py

```diff
@@ -0,0 +1,114 @@
+import unittest
+import itertools
+
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.core.models.configs import MLPEncoderConfig
+from ray.rllib.core.models.base import STATE_IN, STATE_OUT, ENCODER_OUT
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.test_utils import framework_iterator
+
+_, tf, _ = try_import_tf()
+torch, _ = try_import_torch()
+
+
+class TestMLPEncoders(unittest.TestCase):
+    def test_mlp_encoders(self):
+        """Tests building MLP encoders properly and checks for correct architecture."""
+
+        inputs_dims_configs = [[1], [50]]
+
+        list_of_hidden_layer_dims = [[], [1], [64, 64], [1000, 1000, 1000]]
+
+        hidden_layer_activations = [None, "linear", "relu", "tanh", "swish"]
+
+        hidden_layer_use_layernorms = [False, True]
+
+        output_dims_configs = inputs_dims_configs
+
+        output_activations = hidden_layer_activations
+
+        use_biases = [False, True]
+
+        for permutation in itertools.product(
+            inputs_dims_configs,
+            list_of_hidden_layer_dims,
+            hidden_layer_activations,
+            hidden_layer_use_layernorms,
+            output_activations,
+            output_dims_configs,
+            use_biases,
+        ):
+            (
+                inputs_dims,
+                hidden_layer_dims,
+                hidden_layer_activation,
+                hidden_layer_use_layernorm,
+                output_activation,
+                output_dims,
+                use_bias,
+            ) = permutation
+
+            print(
+                f"Testing ...\n"
+                f"inputs_dim: {inputs_dims}\n"
+                f"hidden_layer_dims: {hidden_layer_dims}\n"
+                f"hidden_layer_activation: {hidden_layer_activation}\n"
+                f"hidden_layer_use_layernorm: {hidden_layer_use_layernorm}\n"
+                f"output_activation: {output_activation}\n"
+                f"output_dims: {output_dims}\n"
+                f"use_bias: {use_bias}\n"
+            )
+
+            config = MLPEncoderConfig(
+                input_dims=inputs_dims,
+                hidden_layer_dims=hidden_layer_dims,
+                output_dims=output_dims,
+                hidden_layer_activation=hidden_layer_activation,
+                hidden_layer_use_layernorm=hidden_layer_use_layernorm,
+                output_activation=output_activation,
+                use_bias=use_bias,
+            )
+
+            # To compare number of params between frameworks.
+            tf_counts = None
```

Comment:
same comments regarding the parameter count. We can add a test util class to do this automatically and just stamp the code in other places.

```
class ParameterCountChecker:
     
     def add_model(self, model: Model, key: str):
           ...
      
     def check(self):
           ...
```

We can put this class in test_util and use it like this:

```
pc = ParameterCountChecker()
for fw in ...:
    ...
    pc.add_model(fw, model)
    pc.check()
```

Reply:
Yup, great idea!

We should also actually write a private for-testing-only helper method that overwrites the weights of say an TorchMLPEncoder with the weights of a corresponding TfMLPEncoder to check whether a forward pass results in exactly the same output results for both frameworks. Only then can we say the architectures are really the same, b/c with the test right now, one still might get the activations wrong and wouldn't see that merely by the param counts.

Reply:
Yeah, I think you can rename this utility class and add what you suggested as well., there is like `pc.check_param_count()` and `pc.check_outputs(input_tensor=...)`